Src/ILGPU/Context.cs

```diff
@@ -384,6 +385,77 @@ public DeviceCollection<TDevice> GetDevices<TDevice>()
                 : new DeviceCollection<TDevice>(new List<Device>());
         }
 
+        /// <summary>
+        /// Attempts to return the most optimal single device.
+        /// </summary>
+        /// <param name="PreferCPU">Always returns CPU device 0.</param>
+        /// <returns>Selected device.</returns>
+        public Device GetBestDevice(bool PreferCPU = false)
+        {
+            if(PreferCPU)
+            {
+                return deviceMapping.TryGetValue(AcceleratorType.CPU, out var devices)
```

Comment:
Question: may we cache this lookup operation in the beginning?

Reply:
QUESTION: Could this be replaced with `GetCPUDevice(0)`?

Reply:
We could cache it against subsequent calls of GetPreferredDevice but caching it in the function for the call at the bottom doesn't matter. I did not add caching because I don't expect this function to be called a lot.

Reply:
I did replace it with GetCPUDevice(0) it was the same code effectively.